xdem/coreg/base.py

```diff
@@ -379,7 +379,252 @@ def _preprocess_coreg_raster_input(
     return ref_dem, tba_dem, inlier_mask, transform, crs
 
 
-# TODO: Re-structure AffineCoreg apply function and move there?
+def _preprocess_coreg_fit_raster_point(
+    raster_elev: NDArrayf | MArrayf | RasterType,
+    point_elev: gpd.GeoDataFrame,
+    inlier_mask: NDArrayb | Mask | None = None,
+    transform: rio.transform.Affine | None = None,
+    crs: rio.crs.CRS | None = None,
+) -> tuple[NDArrayf, gpd.GeoDataFrame, NDArrayb, affine.Affine, rio.crs.CRS]:
+    """Pre-processing and checks of fit for raster-point input."""
+
+    # TODO: Convert to point cloud once class is done
+    if isinstance(raster_elev, gu.Raster):
+        rst_elev = raster_elev.data
+        crs = raster_elev.crs
+        transform = raster_elev.transform
+    else:
+        rst_elev = raster_elev
+        crs = crs
+        transform = transform
+
+    if transform is None:
+        raise ValueError("'transform' must be given if both DEMs are array-like.")
+
+    if crs is None:
+        raise ValueError("'crs' must be given if both DEMs are array-like.")
+
+    # Make sure that the mask has an expected format.
+    if inlier_mask is not None:
+        if isinstance(inlier_mask, Mask):
+            inlier_mask = inlier_mask.data.filled(False).squeeze()
+        else:
+            inlier_mask = np.asarray(inlier_mask).squeeze()
+            assert inlier_mask.dtype == bool, f"Invalid mask dtype: '{inlier_mask.dtype}'. Expected 'bool'"
+
+        if np.all(~inlier_mask):
+            raise ValueError("'inlier_mask' had no inliers.")
+    else:
+        inlier_mask = np.ones(np.shape(rst_elev), dtype=bool)
+
+    # TODO: Convert to point cloud?
+    # Convert geodataframe to vector
+    point_elev = point_elev.to_crs(crs=crs)
+
+    return rst_elev, point_elev, inlier_mask, transform, crs
+
+
+def _preprocess_coreg_fit_point_point(
+    reference_elev: gpd.GeoDataFrame, to_be_aligned_elev: gpd.GeoDataFrame
+) -> tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+    """Pre-processing and checks of fit for point-point input."""
+
+    ref_dem = reference_elev
```

Comment:
Is it really worth renaming into "ref_dem". This is a bit comfusing... Same for "tba_dem" also this one is modified at least. Maybe use ref_elev and tba_elev.

Reply:
Done